foxpuppet/windows/browser/panel_ui/panel_ui.py

```diff
@@ -0,0 +1,187 @@
+
+
+class PanelUI(NavBar):
+    """Handles interaction with Panel UI."""
+
+    if TYPE_CHECKING:
+        from foxpuppet.windows import BrowserWindow
+
+    @staticmethod
+    def create(
+        window: Optional["BrowserWindow"], root: WebElement
+    ) -> Type["PanelUI"] | Any:
+        """Create a Panel UI object.
+
+        Args:
+            window (:py:class:`BrowserWindow`): Window object this region
+                appears in.
+            root
+                (:py:class:`~selenium.webdriver.remote.webelement.WebElement`):
+                WebDriver element object that serves as the root for the
+                Panel UI.
+
+        Returns:
+            :py:class:`PanelUI`: Firefox Panel UI.
+
+        """
+        panel_items: dict = {}
+        _id: str | bool | WebElement | dict = root.get_property("id")
+
+        panel_items.update(PANEL_ITEMS)
+        return panel_items.get(_id, PanelUI)(window, root)
+
+    @property
+    def is_update_available(self) -> bool:
+        """
+        Checks if the Panel UI button indicates a pending Firefox update.
+
+        Returns:
+            bool: True if an update notification (barge) is present, False otherwise.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            barged_status = self.selenium.find_element(
+                *PanelUILocators.PANEL_UI_BUTTON
+            ).get_attribute("barged")
+            return barged_status == "true"
+
+    def open_panel_menu(self) -> None:
+        """
+        Opens the Panel UI menu.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*PanelUILocators.PANEL_UI_BUTTON).click()
+            self.wait.until(
+                EC.presence_of_element_located(*PanelUILocators.PANEL_POPUP),
+                message="Panel UI menu did not open",
+            )
+
+    def open_new_tab(self) -> None:
+        """
+        Opens a new tab using the Panel UI menu.
+        """
+        initial_handles = set(self.selenium.window_handles)
+        self.open_panel_menu()
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*PanelUILocators.NEW_TAB).click()
+            self.wait.until(
+                lambda _: set(self.selenium.window_handles) - initial_handles,
+                message="New Tab did not open",
+            )
+
+    def open_new_window(self) -> None:
+        """
+        Opens a new window using the Panel UI menu.
+        """
+        initial_handles = set(self.selenium.window_handles)
+        self.open_panel_menu()
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*PanelUILocators.NEW_WINDOW).click()
+            self.wait.until(
+                lambda _: set(self.selenium.window_handles) - initial_handles,
+                message="New window did not open",
+            )
+
```

Comment:
I wonder if we should switch the selenium execution to the newly opened window or tab. Would be easy to do. What do you think?